ciw/arrival_node.py

```diff
@@ -127,11 +128,15 @@ def inter_arrival(self, nd, clss, current_time):
             return self.simulation.inter_arrival_times[nd][clss](current_time)
         return self.simulation.inter_arrival_times[nd][clss]()
 
-    def batch_size(self, nd, clss):
+    def batch_size(self, nd, clss, current_time):
         """
         Samples the batch size for next class and node.
         """
-        return self.simulation.batch_sizes[nd][clss]()
+        #return int(round(self.simulation.batch_sizes[nd][clss](getattr(self,'next_event_date',None))))
```

Comment:
You are correct about this being 1 "tick" off: here we are determining the next arrival's batch size using the current time. Instead of using `current_time`, we can use `self.event_dates_dict[nd][clss]` the time that has already been sampled for the next arrival.

How does this sound? The method could then look like:
```
    def batch_size(self, nd, clss, current_time):
        """
        Samples the batch size for next class and node.
        """
        if self.simulation.network.customer_classes[
            clss].batching_distributions[nd-1][0] == "TimeDependent":
            return self.simulation.batch_sizes[nd][clss](
                self.event_dates_dict[nd][clss]
                )
        return self.simulation.batch_sizes[nd][clss]()
```